### PR TITLE
Remove ReceiptOverlay and AmountPurchased dependency

### LINKED ISSUE #384: Remove dependency between ReceiptOverlay, and amountPurchased and price
ticket_repository.dart - useTicket()

### PR DESCRIPTION
**Pull request by fremartini**
closes #384

### CODE REVIEW

File: test/features/ticket/presentation/cubit/tickets_cubit_test.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      skip: 2,
+      expect: () => [
+        const TicketUsing([]),
+        TicketUsed(tReceipt, const []),
+        const TicketsLoaded([]),
+      ],
+    );
+
+    blocTest<TicketsCubit, TicketsState>(
+      'should emit [Using, Error Loaded] when state is Loaded',
```
Comment (TTA777) [inline test/features/ticket/presentation/cubit/tickets_cubit_test.dart:90]: Missing ',' between Error and Loaded?

**Review by TTA777 — CHANGES_REQUESTED** [review]
Minor changes requested

File: lib/features/receipt/presentation/widgets/receipt_list_entry_factory.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import 'package:coffeecard/features/receipt/domain/entities/purchase_receipt.dart';
+import 'package:coffeecard/features/receipt/domain/entities/receipt.dart';
+import 'package:coffeecard/features/receipt/domain/entities/swipe_receipt.dart';
+import 'package:coffeecard/features/receipt/presentation/widgets/list_entry/placeholder_receipt_list_entry.dart';
+import 'package:coffeecard/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart';
+import 'package:coffeecard/features/receipt/presentation/widgets/list_entry/swipe_receipt_list_entry.dart';
+import 'package:flutter/material.dart';
+
+abstract class ReceiptListEntryFactory {
+  static Widget create(Receipt receipt) {
```
[the remark this answers is not in the document]
Comment (fremartini) [reply lib/features/receipt/presentation/widgets/receipt_list_entry_factory.dart:11]: no, I don't believe a factory implementation would work. See payment_handler for a similar implimentation; the list_entry widgets also don't inherit from receipt_list_entry so  you can't use a factory here

**Review by TTA777 — APPROVED** [review]
Lgtm for now. Just need to rebase, and fix any of the errors the new linter gives you

**Pull request merged**